src/keygen/Main.hs

```diff
@@ -1,121 +1,87 @@
-    , koGenesisFile  :: FilePath
-    , koStakeholders :: Word
-    , koRichmen      :: Word
-    , koTotalStake   :: Word64
-    }
-
-generateKeyfile :: FilePath -> IO (PublicKey, VssCertificate)
-generateKeyfile fp = do
-    initializeUserSecret fp
-    sk <- snd <$> keyGen
-    vss <- vssKeyGen
-    us <- takeUserSecret fp
-    writeUserSecretRelease $
-        us & usPrimKey .~ Just sk
-           & usVss .~ Just vss
-    expiry <- fromIntegral <$> randomRIO (vssMinTTL :: Int, vssMaxTTL)
-    let vssPk = asBinary $ toVssPublicKey vss
-        vssCert = mkVssCertificate sk vssPk expiry
-    return (toPublic sk, vssCert)
+import           Avvm                 (genGenesis, getHolderId)
+import           KeygenOptions        (AvvmStakeOptions (..), KeygenOptions (..),
+                                       TestStakeOptions (..), optsInfo)
+import           Testnet              (genTestnetStakes, generateKeyfile)
 
 replace :: FilePath -> FilePath -> FilePath -> FilePath
 replace a b = toString . (T.replace `on` toText) a b . toText
 
-optsParser :: Parser KeygenOptions
-optsParser = do
-    koPattern <- strOption $
-        long    "file-pattern" <>
-        short   'f' <>
-        metavar "PATTERN" <>
-        help    "Filename pattern for generated keyfiles \
-                \(`{}` is a place for number)"
-    koGenesisFile <- strOption $
-        long    "genesis-file" <>
-        metavar "FILE" <>
-        value   "genesis.bin" <>
-        help    "File to dump binary shared genesis data"
-    koStakeholders <- option auto $
-        long    "total-stakeholders" <>
-        short   'n' <>
-        metavar "INT" <>
-        help    "Total number of keyfiles to generate"
-    koRichmen <- option auto $
-        long    "richmen" <>
-        short   'm' <>
-        metavar "INT" <>
-        help    "Number of richmen among stakeholders"
-    koTotalStake <- option auto $
-        long    "total-stake" <>
-        metavar "INT" <>
-        help    "Total coins in genesis"
-    pure KO{..}
-
-optsInfo :: ParserInfo KeygenOptions
-optsInfo = info (helper <*> optsParser) $
-    fullDesc `mappend` progDesc "Tool to generate keyfiles"
-
-main :: IO ()
-main = do
-    KO {..} <- execParser optsInfo
-    let keysDir = takeDirectory koPattern
-        genFileDir = takeDirectory koGenesisFile
+getTestnetGenesis :: Coin -> TestStakeOptions -> IO GenesisData
+getTestnetGenesis avvmStake tso@TSO{..} = do
+    let keysDir = takeDirectory tsoPattern
     createDirectoryIfMissing True keysDir
-    createDirectoryIfMissing True genFileDir
 
-    genesisList <- forM [1..koStakeholders] $ \i ->
-        generateKeyfile $ replace "{}" (show i) koPattern
-    print $ show koStakeholders ++ " keyfiles are generated"
+    let totalStakeholders = tsoRichmen + tsoPoors
+    genesisList <- forM [1 .. totalStakeholders] $ \i ->
+        generateKeyfile $ replace "{}" (show i) tsoPattern
+    print $ show totalStakeholders ++ " keyfiles are generated"
```

Comment:
`putText $ show totalStakeholders <> " keyfiles are generated"`, no need for `print` (which adds extra quotes)